jade/wsgi/routes/v1/judgements.py

```diff
@@ -2,49 +2,91 @@
+    def set_judgement_preference(judgement_id, gu_id, request_values):
+        """
+        Sets the preference bit for a specific judgement and returns the event
+        """
+        preference = json.loads(request_values['preference'])
+        if preference not in (True, False):
+            raise errors.ParameterError(
+                "'preference' must be either 'true' or 'false'")
+
+        # Construct a new proto-event for the judgement
+        proto_event = types.events.JudgementPreferenceSet.proto(
+            gu_id=gu_id, judgement_id=judgement_id, preference=preference)
+
+        # Execute the proto_event and construct full event.
+        return util.execute_and_log_or_error(state, proto_event)
+
+    @bp.route("/v1/judgements/", methods=["post"])
+    @util.authorized_user_action(
+        config, trusted_clients, centralauth, "new_judgement")
+    def new_judgement(gu_id, request_values):
+        """Creates a new judgement and returns the event"""
 
-    @bp.route("/judgements/")
-    @authorized_user_action(trusted_clients)
-    def post_judgement(gu_id, request_values):
-        """Creates a new judgement and returns event"""
         context = request_values['context']
-        schema_id = request_values['schema_id']
-        data = json.loads(flask.request.valeus['data'])
-
-        # Checks that data represents a valid JSON blob based on latest schema
-        # version
-        source.validate(schema_id, data)
-
-        # Constructs a new Judgement, inserts it into a DB and acquires
-        # identifiers unique identifier.  This function call does not complete
-        # unless the event is successfully included in the bin log.
-        event = source.new_judgement(gu_id, context, schema_id, data)
-        return flask.json.jsonify(event)
-
-
-def authorized_user_action(trusted_clients):
-    """
-    Wrap a flask route. Ensures that the user has authorized via OAuth or that
-    the request came from an authorized MediaWiki installation and provided
-    a `global_user_id`.  If neither, return an authorization error.
-    """
-    def decorator(route):
-        @wraps(route)
-        def authorized_route(*args, **kwargs):
-            if 'authorization_key' in flask.request:
-                values = trusted_clients.authorize_and_decode(flask.request)
-                gu_id = values['gu_id']
-                return route(gu_id, values, *args, **kwargs)
-            elif 'mwoauth_access_token' in flask.session:
-                gu_id = flask.session.get('mwoauth_identity')['id']
-                return route(gu_id, flask.request.values, *args, **kwargs)
-            else:
-                return responses.auth_error()
-
-    return decorator
+        entity_type = request_values['entity_type']
+        entity_id = request_values['entity_id']
+        schema = request_values['schema']
+        data = json.loads(request_values['data'])
+
+        # Check that data represents a valid JSON blob based on the latest
```

Comment:
Comments are usually bad sign and needed when the code is not explanatory enough, this comment is redundant because "state.schemas.validate(schema, data)" conveys the comment fully.